nbt/src/main/java/net/kyori/adventure/nbt/CompoundTagSetter.java

```diff
@@ -42,6 +44,27 @@
    */
   @NonNull R put(final @NonNull String key, final @NonNull BinaryTag tag);
 
+  /**
+   * Removes a tag.
+   *
+   * @param key the key
+   * @return a compound tag
+   * @since 4.0.0
+   */
+  default @NonNull R remove(final @NonNull String key) {
+    return this.remove(key, null);
+  }
+
+  /**
+   * Removes a tag.
+   *
+   * @param key the key
+   * @param removed a consumer that accepts the removed tag
+   * @return a compound tag
+   * @since 4.0.0
+   */
+  @NonNull R remove(final @NonNull String key, final @Nullable Consumer<BinaryTag> removed);
```

Comment:
Consider making this `@NonNull`, and delegating with `this.remove(key, r -> {});` instead since the lambda should effectively be a constant, and there would be no branch.

Reply:
Is `@Nullable` so that people don't need to do something like:
```java
if(result != null) {
  compound.remove("foo", result);
} else {
  compound.remove("foo");
}
```
etc

Reply:
The [same method](https://github.com/KyoriPowered/adventure/blob/master/nbt/src/main/java/net/kyori/adventure/nbt/ListBinaryTag.java#L147) in `ListBinaryTag` accepts a `@Nullable` `Consumer`, so we will do the same here for consistency.